src/app/rpmostree-compose-builtin-tree.cxx

```diff
@@ -1468,6 +1468,15 @@ rpmostree_compose_builtin_extensions (int             argc,
   if (!repo)
     return FALSE;
 
+  rust::String treeref;
+  if (!opt_extensions_base_rev)
+    {
+      treeref = treefile->get_treeref();
+      if (strlen (treeref.c_str()) == 0)
```

Comment:
You should think Rust/C++ strings here, not C :smile: 
```suggestion
      if (treeref.length() == 0)
```
(The length is already stored, we don't need to recompute it)